cmd/soroban-cli/src/commands/contract/info/interface.rs

```diff
@@ -0,0 +1,55 @@
+use std::fmt::Debug;
+
+use crate::commands::contract::info::interface::Error::NoInterfacePresent;
+use crate::commands::contract::info::shared;
+use crate::commands::contract::info::shared::fetch_wasm;
+use clap::{command, Parser};
+use soroban_spec_tools::contract;
+use soroban_spec_tools::contract::Spec;
+
+use crate::commands::contract::InfoOutput;
+
+#[derive(Parser, Debug, Clone)]
+pub struct Cmd {
+    #[command(flatten)]
+    pub common: shared::Args,
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error(transparent)]
+    Wasm(#[from] shared::Error),
+    #[error(transparent)]
+    Spec(#[from] contract::Error),
+    #[error("no interface present in provided WASM file")]
+    NoInterfacePresent(),
+    #[error(transparent)]
+    Json(#[from] serde_json::Error),
+}
+
+impl Cmd {
+    pub async fn run(&self) -> Result<String, Error> {
+        let bytes = fetch_wasm(&self.common).await?;
+
+        let (base64, spec) = if bytes.is_none() {
+            Spec::spec_to_base64(&soroban_sdk::token::StellarAssetSpec::spec_xdr())?
+        } else {
+            let spec = Spec::new(&bytes.unwrap())?;
+
+            if spec.env_meta_base64.is_none() {
+                return Err(NoInterfacePresent());
+            }
+
+            (spec.spec_base64.unwrap(), spec.spec)
+        };
+
+        let res = match self.common.output {
+            InfoOutput::XdrBase64 => base64,
+            InfoOutput::Json => serde_json::to_string(&spec)?,
+            InfoOutput::JsonFormatted => serde_json::to_string_pretty(&spec)?,
```

Comment:
I like the idea of using the raw XDR<>JSON for the interface format.

There is actually another JSON format we invented at some point, that is visible with this command:
```
❯ stellar contract bindings json --wasm test_empty.wasm
[
  {
    "type": "function",
    "doc": "",
    "name": "empty",
    "inputs": [],
    "outputs": []
  }
]
```

The json command really doesn't belong in the bindings command, because bindings is for generating actual clients, and the JSON is clearly not a client. So I think we can deprecate the `bindings json` command.

The question becomes should we keep this JSON interface format around, or abandon it for the XDR/JSON.

Realistically the two formats don't look _that_ different, and so the value in maintaining the other format that I'm not sure many people know exists has that much value.

```
$ wasm-cs test_empty.wasm read contractspecv0 -f binary | stellar xdr decode --input stream --type ScSpecEntry --output json-formatted
{
  "function_v0": {
    "doc": "",
    "name": "empty",
    "inputs": [],
    "outputs": []
  }
}
```

Reply:
Not sure which wasm file was it, but shouldn't you be able to do the same thing as in 2nd command with `info interface --output json-formatted` ?
Also agree on keeping 1 unified standard, if anybody is using outputs JSONs and parsing it, IMO we 100% want to keep it same across the commands (unless we really need to make it different) 

https://github.com/stellar/stellar-cli/issues/1535